enterprise/wsproxy/wsproxy.go

```diff
@@ -553,6 +567,34 @@ func pingSiblingReplicas(ctx context.Context, logger slog.Logger, sf *singleflig
 	return errStrInterface.(string)
 }
 
+// pingReplica pings a replica over it's internal relay address to ensure it's
+// reachable and alive for health purposes. It will try to ping the replica
+// twice if the first ping fails, with a short delay between attempts.
+func pingReplica(ctx context.Context, client http.Client, replica codersdk.Replica) error {
+	const attempts = 2
+	var err error
+	for i := 0; i < attempts; i++ {
+		err = replicasync.PingPeerReplica(ctx, client, replica.RelayAddress)
+		if err == nil {
+			return nil
+		}
+		if i < attempts-1 {
+			// Use a timer to avoid a goroutine remaining after the test. Newer
+			// versions of Go will garbage collect unused timers, but the
+			// goroutine may still be running after the test completes.
+			timer := time.NewTimer(1 * time.Second)
```

Comment:
We use go 1.24.6, so I'm not sure why we are using a `Timer` here instead of `After`. It seems like you are claiming that `After()` leaves a goroutine running?

This doesn't square with the Go stdlib docs that are pretty unequivocal: "There is no reason to prefer NewTimer when After will do." https://pkg.go.dev/time#After

Reply:
On further investigation timers aren't actually goroutines, TIL! Switched to time.After